test/src/String/test_move.cpp

```diff
@@ -37,14 +37,10 @@ TEST_CASE("Testing String move assignment", "[String-move-02]")
 
 TEST_CASE("Testing String move self assignment", "[String-move-03]")
 {
-#if (defined(GCC_VERSION) && GCC_VERSION >= 13) || (defined(__clang_major__) && __clang_major__ >= 14)
-#pragma GCC diagnostic push
-#pragma GCC diagnostic ignored "-Wself-move"
-#endif
-    arduino::String a("src");
-    a = std::move(a);
-    REQUIRE(a == "src");
-#if defined(GCC_VERSION) && GCC_VERSION >= 13
-#pragma GCC diagnostic pop
-#endif
+    String b;

```

Comment:
While this surely makes the test pass, I'm quite sure the test is not about `self assignment` anymore :upside_down_face: If there's an issue with memory leaking from this we should solve it at core level

Reply:
I missed the point of this test case, I'll drop this commit. I'll look if I can improve the preprocessor checks, since I am failing this locally.

Reply:
According to [Gcc online docs](https://gcc.gnu.org/onlinedocs/cpp/Common-Predefined-Macros.html) at 3.7.2 `GCC_VERSION` is not defined in the way we are using it. I think we could check against `__GNUC__` only. 

I don't understand why on my local machine `GCC_VERSION` is not defined, but `__GNUC__` is.